src/gui/folder.cpp

```diff
@@ -1218,6 +1223,74 @@ void Folder::slotHydrationDone()
     emit syncStateChange();
 }
 
+void Folder::slotNewShellExtensionConnection()
```

Comment:
@mgallien I can do that. Did not do it initially as we already have similar code here for sync states and my code is quite small so making a new class seemed to be an overkill.

Reply:
I still think that the code related to the communication via the local server should be put into a class external to `FolderMan` not only because that is unrelated to managing the set of sync folders but also to have smaller more focused classes (and we already have quite a few big monsters in our current code)